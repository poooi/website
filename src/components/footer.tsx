import { IconProp } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import React from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components/macro'

import { faDiscord } from '@fortawesome/free-brands-svg-icons/faDiscord'
import { faGithub } from '@fortawesome/free-brands-svg-icons/faGithub'
import { faTelegram } from '@fortawesome/free-brands-svg-icons/faTelegram'
import { faWeibo } from '@fortawesome/free-brands-svg-icons/faWeibo'
import { faHeart } from '@fortawesome/free-solid-svg-icons/faHeart'

import { getExactLanguage } from '../utils'

const Icon = ({ icon }: { icon: IconProp }) => (
  <FontAwesomeIcon icon={icon} size="lg" fixedWidth />
)

const Container = styled.div`
  text-align: center;
  min-height: 50px;
  line-height: 50px;
  margin-bottom: 1em;

  a {
    margin: 0 1em;
    transition: 0.3s;
    cursor: pointer;
    white-space: nowrap;
  }
`

const Copyright = styled.span`
  margin-right: 4em;
  white-space: nowrap;
`

export const Footer = () => {
  const { t, i18n } = useTranslation()

  const { language } = i18n
  const lang = getExactLanguage(language)

  return (
    <Container>
      <Copyright>{`Â© ${new Date().getFullYear()} poi Contributors`}</Copyright>
      {['zh-Hans', 'zh-Hant'].includes(lang) ? (
        <>
          <a href="http://weibo.com/letspoi">
            <Icon icon={faWeibo} />
            {t('weibo')}
          </a>
          <a href={t('telegram-group-link')}>
            <Icon icon={faTelegram} />
            {t('telegram')}
          </a>
        </>
      ) : (
        <a href={t('discord-channel-link')}>
          <Icon icon={faDiscord} />
          {t('Discord sub-channel')}
        </a>
      )}
      <a href="https://github.com/poooi/poi">
        <Icon icon={faGithub} />
        {t('github')}
      </a>
      <a href="https://opencollective.com/poi">
        <Icon icon={faHeart} />
        {t('opencollective')}
      </a>
    </Container>
  )
}
