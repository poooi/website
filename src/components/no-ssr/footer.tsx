import { IconProp } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import React from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components/macro'
import { AnchorButton, ButtonGroup, Intent } from '@blueprintjs/core'

import { faDiscord } from '@fortawesome/free-brands-svg-icons/faDiscord'
import { faGithub } from '@fortawesome/free-brands-svg-icons/faGithub'
import { faTelegram } from '@fortawesome/free-brands-svg-icons/faTelegram'
import { faWeibo } from '@fortawesome/free-brands-svg-icons/faWeibo'
import { faHeart } from '@fortawesome/free-solid-svg-icons/faHeart'

import { getExactLanguage } from '../../utils'

const Icon = ({ icon }: { icon: IconProp }) => (
  <FontAwesomeIcon icon={icon} size="lg" fixedWidth />
)

const Container = styled.div`
  text-align: center;
  min-height: 50px;

  a {
    cursor: pointer;
    white-space: nowrap;
  }
`

const Copyright = styled.span`
  margin-right: 4em;
  white-space: nowrap;
`

export const Footer = () => {
  const { t, i18n } = useTranslation()

  const { language } = i18n
  const lang = getExactLanguage(language)

  return (
    <Container>
      <Copyright>{`Â© ${new Date().getFullYear()} poi Contributors`}</Copyright>
      <ButtonGroup minimal>
        {['zh-Hans', 'zh-Hant'].includes(lang) ? (
          <>
            <AnchorButton href="http://weibo.com/letspoi">
              <Icon icon={faWeibo} />
              {t('weibo')}
            </AnchorButton>
            <AnchorButton href={t('telegram-group-link')}>
              <Icon icon={faTelegram} />
              {t('telegram')}
            </AnchorButton>
          </>
        ) : (
          <AnchorButton href={t('discord-channel-link')}>
            <Icon icon={faDiscord} />
            {t('Discord sub-channel')}
          </AnchorButton>
        )}
        <AnchorButton href="https://github.com/poooi/poi">
          <Icon icon={faGithub} />
          {t('github')}
        </AnchorButton>
        <AnchorButton href="https://opencollective.com/poi">
          <Icon icon={faHeart} />
          {t('opencollective')}
        </AnchorButton>
      </ButtonGroup>
    </Container>
  )
}
