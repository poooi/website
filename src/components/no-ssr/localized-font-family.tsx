import { useTranslation } from 'react-i18next'
import { createGlobalStyle } from 'styled-components'

import Head from 'next/head'
import { getExactLanguage } from '../../utils'

const FontFamily = createGlobalStyle<{ $family: string }>`
  body {
    font-family: ${(props) => props.$family}, system-ui, sans-serif;
  }
`

const resources = {
  en: `'Open Sans'`,
  fr: `'Open Sans'`,
  ja: `'Open Sans', 'Noto Sans JP'`,
  'zh-Hans': `'Open Sans', 'Noto Sans SC'`,
  'zh-Hant': `'Open Sans', 'Noto Sans TC'`,
}

const googleFontlink = {
  en: 'https://fonts.googleapis.com/css2?family=Open+Sans&display=swap',
  fr: 'https://fonts.googleapis.com/css2?family=Open+Sans&display=swap',
  ja:
    'https://fonts.googleapis.com/css2?family=Noto+Sans+JP&family=Open+Sans&display=swap',
  'zh-Hans':
    'https://fonts.googleapis.com/css2?family=Noto+Sans+SC&family=Open+Sans&display=swap',
  'zh-Hant':
    'https://fonts.googleapis.com/css2?family=Noto+Sans+TC&family=Open+Sans&display=swap',
}

export const LocalizedFontFamily = () => {
  const { i18n } = useTranslation()

  const { language } = i18n
  const lang = getExactLanguage(language)

  return (
    <>
      <FontFamily $family={resources[lang as keyof typeof resources]} />
      <Head>
        <link
          href={googleFontlink[lang as keyof typeof googleFontlink]}
          rel="stylesheet"
        />
      </Head>
    </>
  )
}
