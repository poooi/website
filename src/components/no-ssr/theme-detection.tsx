import { useContext, useEffect } from 'react'
import { DispatchThemeChangeContext } from '../../theme'

const getDefaultIsDark = (): boolean => {
  if (window.localStorage?.getItem('theme')) {
    return localStorage.getItem('theme') === 'dark'
  }
  return (
    window.matchMedia &&
    window.matchMedia('(prefers-color-scheme: dark)').matches
  )
}

export const ThemeDetection = () => {
  const dispatch = useContext(DispatchThemeChangeContext)

  useEffect(() => dispatch(getDefaultIsDark()), [])

  return null
}
