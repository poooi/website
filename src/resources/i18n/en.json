{
  "language": "English",
  "name": "poi",
  "description": "Scalable KanColle browser and tool.",
  "github": "GitHub",
  "telegram": "Telegram",
  "opencollective": "OpenCollective",
  "weibo": "Weibo",
  "download": "Download {{version}}",
  "linux-x64": "Linux (x64)",
  "linux-deb-x64": "Debian / Ubuntu (x64)",
  "linux-rpm-x64": "Fedora / CentOS (x64)",
  "macos-x64": "macOS (x64)",
  "win-ia32": "Windows (x86) Portable",
  "win-ia32-setup": "Windows (x86)",
  "win-x64": "Windows (x64) Portable",
  "win-x64-setup": "Windows (x64)",
  "stable-hint": "Stable, for most users",
  "beta-hint": "Beta, for new features",
  "other-versions": "Check the complete list",
  "telegram-group-link": "https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ",
  "discord-channel-link": "https://discordapp.com/channels/118339803660943369/367575898313981952",
  "Discord sub-channel": "Discord",
  "Choose another platform": "Choose another platform",
  "lsc": "Large Scale Constructing...",
  "current-theme": "Theme ",
  "Chibaheit": "Chibaheit",
  "Lilywhite": "Lilywhite",
  "Now Loading": "Now Loading",
  "Stable": "Stable",
  "Beta": "Beta",
  "Others": "Others",
  "HostedAt": "Hosted at {{site}}",
  "Old versions": "Old versions",
  "Windows nightlies": "Windows nightlies",
  "Linux and macOS nightlies": "Linux and macOS nightlies",
  "Source code": "Source code",
  "Close dialog": "Close dialog",
  "error-message": "Ah, the webpage has become üê¢. We will fix it.",
  "Reload": "Reload",
  "Downloads": "Downloads",
  "no-content": "We're working on it, please check later."
}
