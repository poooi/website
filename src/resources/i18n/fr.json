{
  "language": "fran√ßais",
  "name": "poi",
  "description": "Navigateur et outil √©volutif pour KanColle",
  "github": "GitHub",
  "telegram": "Telegram",
  "opencollective": "OpenCollective",
  "weibo": "Weibo",
  "download": "T√©l√©chargez {{version}}",
  "linux-x64": "Linux (x64)",
  "linux-deb-x64": "Debian / Ubuntu (x64)",
  "linux-rpm-x64": "Fedora / CentOS (x64)",
  "macos-x64": "macOS (x64)",
  "win-ia32": "Windows (x86) Portable",
  "win-ia32-setup": "Windows (x86)",
  "win-x64": "Windows (x64) Portable",
  "win-x64-setup": "Windows (x64)",
  "stable-hint": "Stable, pour la plupart des utilisateurs",
  "beta-hint": "Beta, pour les nouvelles fonctionalit√©s",
  "telegram-group-link": "https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ",
  "discord-channel-link": "https://discordapp.com/channels/118339803660943369/367575898313981952",
  "Discord sub-channel": "Discord",
  "Choose another platform": "Choisissez une autre plateforme",
  "lsc": "Construction √† grande √©chelle...",
  "current-theme": "Th√®me ",
  "Chibaheit": "Chibaheit",
  "Lilywhite": "Lilywhite",
  "Now Loading": "En cours de chargement",
  "Stable": "Stable",
  "Beta": "Beta",
  "Others": "Autres",
  "HostedAt": "H√©berg√© sur {{site}}",
  "Old versions": "Anciennes versions",
  "Windows nightlies": "Nightlies Windows",
  "Linux and macOS nightlies": "Nightlies Linux et macOS",
  "Source code": "Code source",
  "Close dialog": "Fermer le dialogue",
  "error-message": "Ah, la page Web est devenue üê¢. Nous allons le r√©parer.",
  "Reload": "Rechargez",
  "Downloads": "T√©l√©chargements",
  "no-content": "Nous y travaillons, veuillez v√©rifier plus tard."
}
