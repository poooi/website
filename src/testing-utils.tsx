/* eslint-disable import/no-extraneous-dependencies */

import { render, RenderResult } from '@testing-library/react'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components/macro'
import { lightTheme } from './theme'

export const renderWithTheme = (node: ReactElement): RenderResult => {
  const result = render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
  )
  return {
    ...result,
    rerender: (newNode: ReactElement) =>
      result.rerender(
        <ThemeProvider theme={lightTheme}>{newNode}</ThemeProvider>,
      ),
  }
}
