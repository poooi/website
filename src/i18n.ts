import i18next from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { initReactI18next } from 'react-i18next'

import en from './resources/i18n/en.json'
import fr from './resources/i18n/fr.json'
import ja from './resources/i18n/ja.json'
import zhHans from './resources/i18n/zh-hans.json'
import zhHant from './resources/i18n/zh-hant.json'

export const resources = {
  en: { translation: en },
  fr: { translation: fr },
  ja: { translation: ja },
  'zh-Hans': { translation: zhHans },
  'zh-Hant': { translation: zhHant },
}

const SFallback = ['zh-Hans', 'zh-Hant', 'en']
const TFallback = ['zh-Hant', 'zh-Hans', 'en']

export const languageFallback = {
  default: ['en'],
  'zh-CN': SFallback,
  'zh-HK': TFallback,
  'zh-MO': TFallback,
  'zh-MY': SFallback,
  'zh-SG': SFallback,
  'zh-TW': TFallback,
}

export const i18n = i18next.createInstance()

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: process.env.NODE_ENV === 'development',
    fallbackLng: languageFallback,
    interpolation: {
      escapeValue: false,
    },
    resources,
  })
