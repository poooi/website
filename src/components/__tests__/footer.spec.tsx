import { i18n } from '../../i18n'

import { renderWithTheme } from '../../testing-utils'
import { Footer } from '../no-ssr/footer'

describe('<Footer />', () => {
  it('renders with non zh', () => {
    i18n.changeLanguage('en-US')
    const { asFragment } = renderWithTheme(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders with zh', () => {
    i18n.changeLanguage('zh')
    const { asFragment } = renderWithTheme(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
