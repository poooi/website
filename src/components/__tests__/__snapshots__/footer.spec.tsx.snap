// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders with non zh 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  min-height: 50px;
  line-height: 50px;
  margin-bottom: 1em;
  color: #1f1f1f;
}

.c0 a {
  margin: 0 1em;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  white-space: nowrap;
}

.c0 a:hover {
  color: #106BA3;
}

.c1 {
  margin-right: 4em;
  white-space: nowrap;
}

<div
    class="c0"
  >
    <span
      class="c1"
    >
      © 2019 poi Contributors
    </span>
    <a
      href="https://discordapp.com/channels/118339803660943369/367575898313981952"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-discord fa-w-14 fa-fw fa-lg "
        data-icon="discord"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M297.216 243.2c0 15.616-11.52 28.416-26.112 28.416-14.336 0-26.112-12.8-26.112-28.416s11.52-28.416 26.112-28.416c14.592 0 26.112 12.8 26.112 28.416zm-119.552-28.416c-14.592 0-26.112 12.8-26.112 28.416s11.776 28.416 26.112 28.416c14.592 0 26.112-12.8 26.112-28.416.256-15.616-11.52-28.416-26.112-28.416zM448 52.736V512c-64.494-56.994-43.868-38.128-118.784-107.776l13.568 47.36H52.48C23.552 451.584 0 428.032 0 398.848V52.736C0 23.552 23.552 0 52.48 0h343.04C424.448 0 448 23.552 448 52.736zm-72.96 242.688c0-82.432-36.864-149.248-36.864-149.248-36.864-27.648-71.936-26.88-71.936-26.88l-3.584 4.096c43.52 13.312 63.744 32.512 63.744 32.512-60.811-33.329-132.244-33.335-191.232-7.424-9.472 4.352-15.104 7.424-15.104 7.424s21.248-20.224 67.328-33.536l-2.56-3.072s-35.072-.768-71.936 26.88c0 0-36.864 66.816-36.864 149.248 0 0 21.504 37.12 78.08 38.912 0 0 9.472-11.52 17.152-21.248-32.512-9.728-44.8-30.208-44.8-30.208 3.766 2.636 9.976 6.053 10.496 6.4 43.21 24.198 104.588 32.126 159.744 8.96 8.96-3.328 18.944-8.192 29.44-15.104 0 0-12.8 20.992-46.336 30.464 7.68 9.728 16.896 20.736 16.896 20.736 56.576-1.792 78.336-38.912 78.336-38.912z"
          fill="currentColor"
        />
      </svg>
      Discord
    </a>
    <a
      href="https://github.com/poooi/poi"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-github fa-w-16 fa-fw fa-lg "
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
        />
      </svg>
      GitHub
    </a>
    <a
      href="https://opencollective.com/poi"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-heart fa-w-16 fa-fw fa-lg "
        data-icon="heart"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
          fill="currentColor"
        />
      </svg>
      OpenCollective
    </a>
  </div>
</DocumentFragment>
`;

exports[`<Footer /> renders with zh 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
  min-height: 50px;
  line-height: 50px;
  margin-bottom: 1em;
  color: #1f1f1f;
}

.c0 a {
  margin: 0 1em;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  white-space: nowrap;
}

.c0 a:hover {
  color: #106BA3;
}

.c1 {
  margin-right: 4em;
  white-space: nowrap;
}

<div
    class="c0"
  >
    <span
      class="c1"
    >
      © 2019 poi Contributors
    </span>
    <a
      href="http://weibo.com/letspoi"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-weibo fa-w-16 fa-fw fa-lg "
        data-icon="weibo"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M407 177.6c7.6-24-13.4-46.8-37.4-41.7-22 4.8-28.8-28.1-7.1-32.8 50.1-10.9 92.3 37.1 76.5 84.8-6.8 21.2-38.8 10.8-32-10.3zM214.8 446.7C108.5 446.7 0 395.3 0 310.4c0-44.3 28-95.4 76.3-143.7C176 67 279.5 65.8 249.9 161c-4 13.1 12.3 5.7 12.3 6 79.5-33.6 140.5-16.8 114 51.4-3.7 9.4 1.1 10.9 8.3 13.1 135.7 42.3 34.8 215.2-169.7 215.2zm143.7-146.3c-5.4-55.7-78.5-94-163.4-85.7-84.8 8.6-148.8 60.3-143.4 116s78.5 94 163.4 85.7c84.8-8.6 148.8-60.3 143.4-116zM347.9 35.1c-25.9 5.6-16.8 43.7 8.3 38.3 72.3-15.2 134.8 52.8 111.7 124-7.4 24.2 29.1 37 37.4 12 31.9-99.8-55.1-195.9-157.4-174.3zm-78.5 311c-17.1 38.8-66.8 60-109.1 46.3-40.8-13.1-58-53.4-40.3-89.7 17.7-35.4 63.1-55.4 103.4-45.1 42 10.8 63.1 50.2 46 88.5zm-86.3-30c-12.9-5.4-30 .3-38 12.9-8.3 12.9-4.3 28 8.6 34 13.1 6 30.8.3 39.1-12.9 8-13.1 3.7-28.3-9.7-34zm32.6-13.4c-5.1-1.7-11.4.6-14.3 5.4-2.9 5.1-1.4 10.6 3.7 12.9 5.1 2 11.7-.3 14.6-5.4 2.8-5.2 1.1-10.9-4-12.9z"
          fill="currentColor"
        />
      </svg>
      Weibo
    </a>
    <a
      href="https://t.me/joinchat/AoMUpkCr6B8uH7EUewq6eQ"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-telegram fa-w-16 fa-fw fa-lg "
        data-icon="telegram"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm121.8 169.9l-40.7 191.8c-3 13.6-11.1 16.9-22.4 10.5l-62-45.7-29.9 28.8c-3.3 3.3-6.1 6.1-12.5 6.1l4.4-63.1 114.9-103.8c5-4.4-1.1-6.9-7.7-2.5l-142 89.4-61.2-19.1c-13.3-4.2-13.6-13.3 2.8-19.7l239.1-92.2c11.1-4 20.8 2.7 17.2 19.5z"
          fill="currentColor"
        />
      </svg>
      Telegram
    </a>
    <a
      href="https://github.com/poooi/poi"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-github fa-w-16 fa-fw fa-lg "
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
        />
      </svg>
      GitHub
    </a>
    <a
      href="https://opencollective.com/poi"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-heart fa-w-16 fa-fw fa-lg "
        data-icon="heart"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
          fill="currentColor"
        />
      </svg>
      OpenCollective
    </a>
  </div>
</DocumentFragment>
`;
